src/check/check_types/unify.zig

```diff
@@ -608,6 +610,22 @@ const Unifier = struct {
                         try self.unifyGuarded(a_var, b_var);
                         self.merge(vars, vars.b.desc.content);
                     },
+                    .list_unbound => {
+                        // When unifying list with list_unbound, list wins
+                        self.merge(vars, vars.a.desc.content);
+                    },
+                    else => return error.TypeMismatch,
+                }
+            },
+            .list_unbound => {
+                switch (b_flat_type) {
+                    .list => |_| {
+                        // When unifying list_unbound with list, list wins
+                        self.merge(vars, vars.b.desc.content);
+                    },
+                    .list_unbound => {
+                        // Both are list_unbound - stay unbound, no merge needed
```

Comment:
I think we still need to merge here, to link these `Var`s together (ie to have 1 redirect to the other)